app/components/support_interface/reference_with_feedback_component.rb

```diff
@@ -34,12 +34,24 @@ def rows
         consent_row,
 
       ].flatten.compact
+
+      base_rows.insert(6, confidentiality_row) if FeatureFlag.active?(:show_reference_confidentiality_status) && reference.feedback_provided?
```

Comment:
`6` feels like a magic number here. 
Given that the array will be flattened and compact, the `confidentiality_row` could return a `nil` making this a little simplier